### PR TITLE
Fix terminating and unequal use of hungarian

### PR DESCRIPTION
**Pull request by RobZelluf**
Terminating is only done when the node was initialized

### CODE REVIEW

File: roboteam_ai/src/bt/Node.cpp [hunk 1]
```diff
@@ -73,10 +74,15 @@ Node::Status Node::NodeUpdate() {
 
 void Node::NodeInitialize() {
     initialize();
+    init = true;
```
Comment (ThijsLuttikhuis) [inline roboteam_ai/src/bt/Node.cpp]: um wat? y u need dis

File: roboteam_ai/src/coach/OffensiveCoach.h [hunk 2]
```diff
@@ -35,21 +35,19 @@ class OffensiveCoach {
...
-
     std::vector<Vector2> getDefaultLocations();
-    std::vector<Vector2> getNewOffensivePositions(int numberOfRobots);
-
-    const set<RobotPtr> &getSideAttackers() const;
+    void updateOffensivePositions();
+    std::vector<Vector2> getOffensivePositions(int numberOfRobots);
 
-    void addSideAttacker(RobotPtr);
+    void addSideAttacker(const RobotPtr&);
```
Comment (ThijsLuttikhuis) [inline roboteam_ai/src/coach/OffensiveCoach.h]: add name of thing

**Review by ThijsLuttikhuis — DISMISSED** [review]
change plox

File: roboteam_ai/src/coach/OffensiveCoach.cpp [hunk 3]
```diff
@@ -92,40 +92,55 @@ std::vector<Vector2> OffensiveCoach::getNewOffensivePositions(int numberOfRobots
...
 }
 
-void OffensiveCoach::removeSideAttacker(const OffensiveCoach::RobotPtr& robot) {
-    int size = sideAttackers.size();
-    for (auto &sideAttacker : sideAttackers) {
-        if(sideAttacker->id == robot->id) {
-            sideAttackers.erase(sideAttacker);
-        }
+std::vector<Vector2> OffensiveCoach::getOffensivePositions(int numberOfRobots) {
+    numberOfRobots += numberOfRobots % 2;
```
Comment (rolfvdhulst) [inline roboteam_ai/src/coach/OffensiveCoach.cpp]: ehh wtf is this black magic. This needs a comment at least

File: roboteam_ai/src/skills/SideAttacker.h [hunk 4]
```diff
@@ -19,7 +19,7 @@ class SideAttacker : public Skill {
     Vector2 deltaPos;
     Vector2 targetPos;
 
-    int zone = -1;
+    int zone;
```
Comment (rolfvdhulst) [inline roboteam_ai/src/skills/SideAttacker.h]: looks like a testing artifact to me that you removed this, was there a good reason?

**Review by rolfvdhulst — CHANGES_REQUESTED** [review]
Looks okay, see comments

File: roboteam_ai/src/coach/OffensiveCoach.cpp (hunk 3 above)
Comment (RobZelluf) [reply roboteam_ai/src/coach/OffensiveCoach.cpp]: done

File: roboteam_ai/src/skills/SideAttacker.h (hunk 4 above)
Comment (RobZelluf) [reply roboteam_ai/src/skills/SideAttacker.h]: Removed it, all this logic is moved to the OffensiveCoach

**Review by rolfvdhulst — APPROVED** [review]

**Pull request merged**